test/index.ts

```diff
@@ -260,6 +260,103 @@ pub mod a;`,
     }
   });
 
+  QUnit.test("generates a typescript wrapper", async (assert) => {
+    const input = await createTempDir();
+    input.write({
+      "Cargo.toml": `
+[package]
+name = "hello_lib"
+version = "0.1.0"
+authors = ["Kris Selden <kris.selden@gmail.com>"]
+
+[lib]
+crate-type = ["cdylib"]
+
+[profile.dev]
+opt-level = 1
+
+[dependencies]
+`,
+      "src": {
+      "lib.rs": `
+#[no_mangle]
+pub extern fn foo() {
+  extern {
+    fn external1();
+    fn external2(a: i32);
+    fn external3() -> i32;
+    fn external4(a: i32, b: u32) -> f64;
+  }
+
+  unsafe {
+    external1();
+    external2(2);
+    external3();
+    external4(1, 2);
+  }
+}
+
+#[no_mangle]
+pub extern fn bar(_a: *const u32) {}
+
+#[no_mangle]
+pub extern fn baz() -> i32 { 3 }
+
+#[no_mangle]
+pub extern fn another(_a: i32, _b: u32) -> i32 { 3 }
+`,
+      },
+    });
+    try {
+      const plugin = new Rust(input.path(), {
+        generateAsyncWrapper: true,
+        generateTypescript: true,
+      });
+      const output = createBuilder(plugin);
+      try {
+        await output.build();
+
+        assert.deepEqual(output.changes(), {
+          "hello_lib.d.ts": "create",
+          "hello_lib.js": "create",
+        });
+        const source = fs.readFileSync(output.path("hello_lib.d.ts"), "utf8");
+        assert.strictEqual(source, `
+export interface FunctionImports {
+  external1(): void;
+  external2(arg0: number): void;
+  external3(): number;
+  external4(arg0: number, arg1: number): number;
+}
+
+export interface Imports {
+  env: FunctionImports;
+}
+
+export class Memory {
+  readonly buffer: ArrayBuffer;
```

Comment:
probably should add `grow(pages: number): number;`

Reply:
In glimmer-vm I added a dependency on `@types/webassembly-js-api` (the npm package) to get this. Should we just use that instead? Or is that too onerous in the sense that it then requires all consumers of this to add such a dependency as well?